Gems/ONNX/Code/Source/Clients/Mnist.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "Mnist.h"
+
+namespace ONNX
+{
+    template<typename T>
+    static void MNIST::softmax(T& input)
+    {
+        float rowmax = *std::max_element(input.begin(), input.end());
+        std::vector<float> y(input.size());
+        float sum = 0.0f;
+        for (size_t i = 0; i != input.size(); ++i)
+        {
+            sum += y[i] = std::exp(input[i] - rowmax);
+        }
+        for (size_t i = 0; i != input.size(); ++i)
+        {
+            input[i] = y[i] / sum;
+        }
+    }
+
+    std::ptrdiff_t MNIST::GetResult()
+    {
+        softmax(m_output);
+        m_result = std::distance(m_output.begin(), std::max_element(m_output.begin(), m_output.end()));
+        return m_result;
+    }
+
+    void MNIST::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        Run(m_input, m_output);
+        DispatchTimingSample();
+    }
+
+    void MNIST::DispatchTimingSample()
+    {
+        // Ignore any unusually large inference times - this is mostly an issue with the first inference on CUDA, where the average runtimes
+        // and ImGui graphs are distorted by anomalies.
+        if (m_delta < 10.0f)
```

Comment:
Let's get rid of that. This was just meant for some nicer graph for the screenshot but let's not try to remove samples from users as they might need to see these outliers for debugging purposes.

Reply:
Committed